### ISSUE TITLE
Pipeline Job: fetch pipeline configuration and template from SCM 

### ISSUE THREAD

**Issue by steven-terrana** [jenkinsci/templating-engine-plugin#100]
### Pipeline Job: fetch pipeline configuration and template from SCM 

Currently, when creating a Pipeline Job and using JTE, it's you can only define the pipeline template and pipeline configuration within the Jenkins UI. 

This feature request is to be able to fetch the template and configuration from an SCM, similar to what can be done for a Governance Tier.

**Comment by anandautam1** [comment]
@steven-terrana When would this feature be available ?
With my current organisation behaviour we do not necessarily use the gitBranch plugin to setup our CI pipeline.
Would it be possible for a single job to expand and have the options to load the library version, which configuratio and which pipeline template to use. Our CI pipeline traceibility is also being questioned so to pin point the version of config and which template wil be great .. Thanks !!

**Comment by cokieffebah** [comment]
@anandautam1 re: "Would it be possible for a single job to expand and have the options to load the library version, which configuratio and which pipeline template to use"
Are you asking for a feature where you 'choose from files in the SCM' the file that represents the pipeline configuration and a similar ability to select the file that represents the pipeline template? 
Also, not sure what you mean by "have the options to load the library version"?
Thanks in advance for the forthcoming response

**Comment by anandautam1** [comment]
Hi @cokieffebah and @steven-terrana 
sorry for the ambiguous request. But hey take 2 let me get the intention straight.

Yeah that is correct, that will be the feature where we choose from files in the SCM the file that represents the pipeline configuration and a similar ability to select the file that represents the pipeline template though expanded for a singular job currently we can only copy and paste plain text code based on a singular job.

The idea is to conveniently switch our sub organization to switch their branch (for the individual job as we do not use gitBranch plugin) rather than copy and pasting the developers code into the singular job configuration which reduce errors when testing their pipeline migration. Thus ensure that everything comes in from the SCM.

So to put it simply we would like the single job to have the same option as the global configuration for the library loading and from which branch on based on the singular job level. Hopefully the diagram below explains further.

![2020-09-12_14h21_32](https://user-images.githubusercontent.com/20267877/92987413-7d0f0780-f505-11ea-8aaa-6ffaeab48edc.png) .. Thanks !!
